frontend/packages/console-app/src/components/tour/TourStepComponent.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import { Popover, PopoverPlacement } from '@console/shared';
+import StepHeader from './steps/StepHeader';
+import StepFooter from './steps/StepFooter';
+import StepBadge from './steps/StepBadge';
+import { TourContext } from './tour-context';
+import { TourActions } from './const';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+import { Spotlight } from './Spotlight';
+import './TourStepComponent.scss';
+
+type TourStepComponentProps = {
+  selector?: string;
+  placement?: string;
+  heading: string;
+  content: React.ReactNode;
+  showClose?: boolean;
+  showStepBadge?: boolean;
+  nextButtonText?: string;
+  backButtonText?: string;
+  onNext?: () => void;
+  onBack?: () => void;
+  onClose?: () => void;
+};
+
+const TourStepComponent: React.FC<TourStepComponentProps> = ({
+  placement,
+  heading,
+  content,
+  selector,
+  showStepBadge = true,
+  showClose = true,
+  nextButtonText,
+  backButtonText,
+  onNext,
+  onBack,
+  onClose,
+}) => {
+  const {
+    tourState: { stepNumber: step },
+    totalSteps,
+    tourDispatch,
+  } = React.useContext(TourContext);
+  const header = <StepHeader>{heading}</StepHeader>;
+  const footer = (
+    <StepFooter
+      primaryButton={{
+        name: nextButtonText ?? 'Next',
+        callback: () => {
+          onNext
+            ? onNext()
+            : step > totalSteps
+            ? tourDispatch(TourActions.complete)
+            : tourDispatch(TourActions.next);
```

Comment:
Why bother with having a very smart component that still has an `onNext` and `onBack` callback for handling unknown cases.

You should split this up into a pure component and then a controlled component for handling the tour steps and then you will still be able to have your close modal component.